From 0fef55014e440d6e30921f7cd9d9f7e738995866 Mon Sep 17 00:00:00 2001
From: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Date: Wed, 2 May 2018 13:29:20 +0300
Subject: [PATCH 3650/5242] MLK-18163-2: drm: imx: dcss: Fix bit depth
 checking for YUV420

commit  a4e796b88a4c23f7a9252c93b6554e54076f7791 from
https://source.codeaurora.org/external/imx/linux-imx.git

When YUV420 is used, we need to check that the deep color mode actually
supports the bit depth required. Currently, the code checks the RGB bit
depth.

Signed-off-by: Laurentiu Palcu <laurentiu.palcu@nxp.com>
(cherry picked from commit 5ac33861906cb7be660cc5c0a0f494194a81275a)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/dcss/dcss-crtc.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/imx/dcss/dcss-crtc.c b/drivers/gpu/drm/imx/dcss/dcss-crtc.c
index 94dcb23..84ed044 100644
--- a/drivers/gpu/drm/imx/dcss/dcss-crtc.c
+++ b/drivers/gpu/drm/imx/dcss/dcss-crtc.c
@@ -191,7 +191,7 @@ void dcss_crtc_setup_opipe(struct drm_crtc *crtc, struct drm_connector *conn,
 	/* FIXME: we should get the connector colorspace some other way */
 	if (vic == 97 &&
 	    (di->color_formats & DRM_COLOR_FORMAT_YCRCB420) &&
-	    (di->bpc >= 10))
+	    (di->hdmi.y420_dc_modes & DRM_EDID_YCBCR420_DC_30))
 		dcss_crtc->opipe_pix_format = DRM_FORMAT_P010;
 	else
 		dcss_crtc->opipe_pix_format = DRM_FORMAT_ARGB8888;
-- 
1.7.9.5

